deploy/local_single_user_deploy.sh

```diff
@@ -0,0 +1,218 @@
+        linked_to_lib_path="$(readlink -f ${lib_path})"
+    done
+}
+
+copy_shared_libs() {
+    libs_visor=$(extract_shared_libraries ${VISOR_BIN})
+    libs_keeper=$(extract_shared_libraries ${KEEPER_BIN})
+    libs_client=$(extract_shared_libraries ${CLIENT_BIN})
+
+    # Combine shared libraries from all executables and remove duplicates
+    all_shared_libs=$(echo -e "${libs_visor}\n${libs_keeper}\n${libs_client}" | sort | uniq)
+
+    # Copy shared libraries to the lib directory
+    echo -e "${DEBUG}Copying shared library ...${NC}"
+    mkdir -p ${LIB_DIR}
+    for lib in ${all_shared_libs}; do
+        if [[ ! -z ${lib} ]]
+        then
+            copy_shared_libs_recursive ${lib} ${LIB_DIR}
+        fi
+    done
+}
+
+# Functions to launch and kill processes
+launch_process() {
+    local bin="$1"
+    local args="$2"
+    local log_file="$3"
+    echo -e "${DEBUG}Launching $bin ...${NC}"
+    LD_LIBRARY_PATH=${LIB_DIR} nohup ${bin} ${args} > ${BIN_DIR}${log_file} 2>&1 &
+}
+
+kill_process() {
+    local bin="$1"
+    echo -e "${DEBUG}Killing $(basename ${bin}) ...${NC}"
+    pkill -9 -f ${bin}
+}
+
+stop_process() {
+    local bin="$1"
+    echo -e "${DEBUG}Stopping $(basename ${bin}) ...${NC}"
+    pkill -f ${bin}
+}
+
+# Main functions for install, reset, and usage
+install() {
+    echo -e "${INFO}Installing ...${NC}"
+    copy_shared_libs
+    check_files
+    generate_config_files ${NUM_KEEPERS} ${BIN_DIR} ${CONF_FILE} ${CONF_DIR}
+    launch_process ${VISOR_BIN} "--config ${CONF_DIR}/visor_conf.json" "/visor.log"
+    num_keepers=${NUM_KEEPERS}
+    for (( i=0; i<num_keepers; i++ ))
+    do
+        #local keeper_args="--config ${CONF_DIR}/keeper_conf_$i.json"
+        launch_process ${KEEPER_BIN} "--config ${CONF_DIR}/keeper_conf_$i.json" "/keeper_$i.log"
+    done
+    launch_process ${CLIENT_BIN} "--config ${CONF_DIR}/client_conf.json" "/client.log"
+    echo -e "${DEBUG}Install done${NC}"
+}
+
+reset() {
+    echo -e "${INFO}Resetting ...${NC}"
+    kill_process ${VISOR_BIN}
+    kill_process ${KEEPER_BIN}
+    kill_process ${CLIENT_BIN}
+    echo -e "${DEBUG}Reset done${NC}"
+}
+
+stop() {
+    echo -e "${INFO}Stopping ...${NC}"
+    stop_process ${VISOR_BIN}
+    stop_process ${KEEPER_BIN}
+    stop_process ${CLIENT_BIN}
+    echo -e "${DEBUG}Stop done${NC}"
+}
+
+# Usage function with new options
+usage() {
+    echo "Usage: $0 [options]"
+    echo "Options:"
+    echo "  -h, --help             Display this help and exit"
+    echo "  -n, --num-keepers NUM  Set the number of keeper processes"
+    echo "  -w, --work-dir DIR     Set the working directory"
+    echo "  -i, --install          Install all components"
+    echo "  -r, --reset            Reset all components"
+    echo "  -s, --stop             Stop all components"
+    exit 1
+}
+
+# Parse arguments to set variables
+parse_args() {
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+            -i|--install) install; shift ;;
```

Comment:
the order of arguments should not matter. if `-i` is passed before before `-n` and `-w`, then `install` will be executed without the modified values for those variables.